### DISCUSSION TITLE
Can llama.cpp support run different attentions layers on different devices?

### DISCUSSION THREAD

**Discussion by chosen-ox** [ggml-org/llama.cpp#10300] [Q&A] [answered]
For example, llama3.2 1b have 16 layers. Can I run the first 8 layers on one machine and then sent kv cache to another machine to run another 8 layers. This is simliar to idea of pipeline parallelism. If not supported now, I wonder can llama.cpp run only partial layers of model or not? If so, I can support pipeline parallelism manually. If llama.cpp can only run the inference as a whole, then I cannot do that.

**Comment by egebeysel** [comment, marked as answer]
I was also looking if llama.cpp, I think this #6017 added pipeline parallelism support.

**Reply by chosen-ox** [reply]
Thank you for your reply. I appreciate it.
